src/main/java/org/isf/opd/service/OpdIoOperationRepositoryImpl.java

```diff
@@ -82,22 +82,23 @@ private TypedQuery<Opd> getOpdQuery(
 					cb.equal(opd.join("ward").get("code"), ward.getCode())
 			);
 		}
-		if (diseaseTypeCode != null) {
+		if (diseaseTypeCode != null && !diseaseTypeCode.equals("")) {
 			predicates.add(
 					cb.equal(opd.join("disease").join("diseaseType").get("code"), diseaseTypeCode)
 			);
 		}
-		if (diseaseCode != null) {
+		if (diseaseCode != null && !diseaseCode.equals("")) {
 			predicates.add(
 					cb.equal(opd.join("disease").get("code"), diseaseCode)
 			);
 		}
-		if (ageFrom != 0 || ageTo != 0) {
+		if (ageFrom > 0 && ageTo < 200) {
```

Comment:
AND operation would cause the input `0 - 199` not entering the if block.

If you don't like `!= 0` and you don't want negative values you may propose (with OR) `if (ageFrom > 0 || ageTo > 0)`
otherwise `ageTo < 200` can still be negative.

Reply:
ageFrom should be greater or equals than 0 and ageFrom should be less than ageTo , ageTo should be less than 200. that is the requiement

Reply:
we are talking about a filter, not the insertion of a new record....